datadog-setup.php

```diff
@@ -47,23 +71,353 @@ function print_help()
 
 Usage:
     Interactive
-        php datadog-setup.php ...
+        php datadog-setup.php [command] ...
     Non-Interactive
         php datadog-setup.php --php-bin php ...
         php datadog-setup.php --php-bin php --php-bin /usr/local/sbin/php-fpm ...
+        php datadog-setup.php config get --php-bin php -d datadog.profiling.enabled
+        php datadog-setup.php config set --php-bin php -d datadog.profiling.enabled=On
 
 Options:
-    -h, --help                  Print this help text and exit
-    --php-bin all|<path to php> Install the library to the specified binary or all php binaries in standard search
-                                paths. The option can be provided multiple times.
+    -h, --help                  Print this help text and exit.
+    --php-bin all|<path to php> Install the library to the specified binary or
+                                all php binaries in standard search paths. The
+                                option can be provided multiple times.
     --install-dir <path>        Install to a specific directory. Default: '/opt/datadog'
-    --uninstall                 Uninstall the library from the specified binaries
+    --uninstall                 Uninstall the library from the specified binaries.
     --enable-appsec             Enable the application security monitoring module.
     --enable-profiling          Enable the BETA profiling module.
+    -d setting[=value]          Used in conjunction with `config <set|get>`
+                                command to specify the INI setting to get or set.
+
+Available commands:
+    config list                 List Datadog's INI setting for the specified binaries.
+    config get                  Get INI setting for the specified binaries.
+    config set                  Set INI setting for the specified binaries.
 
 EOD;
 }
 
+// phpcs:disable PSR1.Classes.ClassDeclaration.MissingNamespace
+class IniRecord
+{
+    /** @var string */
+    public $setting;
+    /** @var string */
+    public $currentValue;
+    /** @var string */
+    public $defaultValue;
+    /** @var string */
+    public $iniFile;
+}
+
+/**
+ * @param array $options
+ * @return Iterator<string, IniRecord>
+ */
+function config_list(array $options)
+{
+    $iniSettings = get_ini_settings('', '', '');
+
+    // The first 3 are 'extension' type of settings.
+    $iniSettings = array_slice($iniSettings, CMD_CONFIG_NUM_SHIFT);
+
+    // Build an index by unique names for filtering.
+    $indexByName = array_column($iniSettings, null, 'name');
+
+    foreach (require_binaries_or_exit($options) as $command => $fullPath) {
+        $binaryForLog = ($command === $fullPath) ? $fullPath : "$command ($fullPath)";
+        echo "Datadog configuration for binary: $binaryForLog", PHP_EOL;
+
+        $iniFilePaths = find_all_ini_files(ini_values($fullPath));
+
+        foreach ($iniFilePaths as $iniFilePath) {
```

Comment:
If multiple binaries point to the same ini, it'll dump the ini files from the same ini multiple times.
This needs to be two separate loops, with the second loop iterating over `array_unique($iniFilePaths);`

```suggestion
    $iniFilePaths = [];
    foreach (require_binaries_or_exit($options) as $command => $fullPath) {
        $binaryForLog = ($command === $fullPath) ? $fullPath : "$command ($fullPath)";
        echo "Datadog configuration for binary: $binaryForLog", PHP_EOL;

        array_push($iniFilePaths, ...find_all_ini_files(ini_values($fullPath));
    }

    foreach (array_unique($iniFilePaths) as $iniFilePath) {
```

Reply:
This would change the output for `php datadog-setup.php config list --php-bin /opt/php/8.2/bin/php --php-bin /opt/php/8.1/bin/php` from:
```bash
datadog.profiling.enabled = On; default: 1, binary: /opt/php/8.2/bin/php, INI file: /opt/php/8.2/etc/conf.d/98-ddtrace.ini
datadog.profiling.enabled = On; default: 1, binary: /opt/php/8.1/bin/php, INI file: /opt/php/8.2/etc/conf.d/98-ddtrace.ini
```
to
```bash
datadog.profiling.enabled = On; default: 1, binary: /opt/php/8.2/bin/php, INI file: /opt/php/8.2/etc/conf.d/98-ddtrace.ini
```
This is also correct, but I would argue that the first output makes it more explicit what happened:
- both PHP binaries are using the same INI file
- in both PHP binaries the INI setting is set to a certain value

It doubles the amount of lines shown, but it is the more explicit approach. If we `array_unique()` the INI files, we should present this information (that multiple binaries share an INI file) somehow to the user.

An example could be something like this:
```
Hint: The INI file /opt/php/8.2/etc/conf.d/98-ddtrace.ini is used for the PHP binaries /opt/php/8.2/bin/php and /opt/php/8.1/bin/php.
datadog.profiling.enabled = On; default: 1, binary: /opt/php/8.2/bin/php, INI file: /opt/php/8.2/etc/conf.d/98-ddtrace.ini
```

This could also make it explicit what happened. I think that if we do not somehow point out this special case, people might get confused about missing INI settings for certain PHP binaries. WDYT?

Reply:
@bwoebi If this is okay for you, do you see something else blocking this PR?